src/coreComponents/common/format/table/TableLayout.hpp

```diff
@@ -51,68 +55,328 @@ class TableLayout
-    Alignment alignment = Alignment::right;
-    /// A boolean to display a colummn
-    bool enabled = true;
-    /// Vector containing substring column name delimited by "\n"
-    std::vector< string > splitColumnNameLines;
+    /// Alignment for column name. By default aligned to center
+    Alignment headerAlignment = Alignment::center;
+    /// Alignment for column values. By default aligned to right side
+    Alignment valueAlignment = Alignment::right;
+  };
+
+/**
+ * @struct CellLayout
+ * @brief Structure representing a cell in a table.
+ * This structure contains information about the cell such as its type, alignment, maximum data length, and width.
+ */
+  struct CellLayout
+  {
+    /// vector containing the cell name separated by a '\n'.
+    std::vector< string > m_lines;
+    /// The type of the cell (Header,Value, Merge, ...).
+    CellType m_cellType;
+    /// The alignment of the cell (left, center, right).
+    Alignment m_alignment;
+    /// Maximum length of the data in the cell.
+    size_t m_maxDataLength;
 
     /**
-     * @brief Construct a ColumnParam object with the specified name and alignment.
-     * @param name The name of the column
-     * @param align The alignment of the column
+     * @brief Constructor to initialize a Cell with a specific type and value.
      */
-    ColumnParam( std::string const & name, Alignment align )
-      : columnName( name ), alignment( align )
-    {}
+    CellLayout();
 
     /**
-     * @brief Construct a ColumnParam object with the specified name, alignment, and display flag.
-     * @param name The name of the column
-     * @param align The alignment of the column
-     * @param display Flag indicating whether the column is enabled
+     * @brief Constructor to initialize a cell given celltype, value and alignment.
+     * @param cellType The type of the cell.
+     * @param value The value to be assigned to the cell.
+     * @param alignment The alignment of the cell (left, right, or center).
      */
-    ColumnParam( std::string const & name, Alignment align, bool display )
-      : columnName( name ), alignment( align ), enabled( display )
-    {}
+    CellLayout( CellType cellType, string const & value, TableLayout::Alignment alignment );
   };
 
   /**
-   * @brief Struct for a column.
+   * @class Column
+   * @brief Class representing a column in a table layout.
+   */
+  class Column
+  {
+public:
+    /// The name of the column.
+    CellLayout m_columName;
+    /// A vector containing all sub-columns in the column.
+    std::vector< Column > m_subColumn;
+    /// struct containing m_alignment for the column (header and values)
+    CellAlignment m_cellAlignment;
+    /// Pointer to the parent column (if any).
+    Column * m_parent;
+    /// Pointer to the next column (if any).
+    Column * m_next;
+
+    /// The width of the cell (e.g., for cell containing subColumns).
+    size_t m_cellWidth = 0;
```

Comment:
1. "`m_columnName`" seems outdated or not suiting here.
2. As we can consider that "*column = header + values*", your `CellAlignement` seem to be more a `ColumnAlignement` structure.
3. I would advise to set pointers default values here, like the other attributes you'd like to always initialize. Pointers are can result in UB, which is ever more dangerous than an unitialized `m_cellWidth`.
4. Is the visibility of all parameter really public here?
5. `m_cellWidth` does not seem well named.

So to sum up what I suggest here:
```suggestion
public:
    /// The header cell layout.
    CellLayout m_header; (note: or m_headerLayout)
    /// A vector containing all sub-columns in the column.
    std::vector< Column > m_subColumn;
    /// struct containing alignment for the column (header and values)
    ColumnAlignment m_alignment;
private:
    /// Pointer to the parent column (nullptr if none).
    Column * m_parent = nullptr;
    /// Pointer to the next column (nullptr if none).
    Column * m_next = nullptr;

    /// The width of the cell (e.g., for cell containing subColumns).
    size_t m_headerMergeCount = 0;
```
(and also renaming the `CellAlignement` struct)

Reply:
PS: if you *really* doesn't want to create accessors for such values as your `m_cellWidth` (which is something like a cache value to work your algorithm on), you can consider the following suggestions:
- adding `TableFormatter` as a friend class of `Column`,
- creating a local struct in `TableTextFormatter::populateHeaderCellsLayout()` that inherit from `Column` and adds-up the needed algorithm local cache values per instance, but I don't think that applies to you case as you need to keep the values along the iterations.

In any case, do not expose internal values like `m_cellWidth` with public qualifier.